### PR TITLE
add register method for User

### PR DESCRIPTION
**Pull request by BohdanTrue**
- create User entity
- create controller/service/repository layer's for User
- create register() method
- create the required DTOs and mapper
- create FieldMatch annotation for checking that password and repeatPassword field in the UserRegistrationRequestDto are the same
- create RegisterException and handle him in GlobalExceptionHandler

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -85,11 +85,6 @@
             <artifactId>spring-boot-starter-validation</artifactId>
         </dependency>
 
-        <dependency>
```
Comment (vovasalyha) [inline pom.xml]: Why did you remove this starter here?

File: src/main/java/mate/academy/bookstore/dto/UserRegistrationRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotNull;
+import lombok.Data;
+import mate.academy.bookstore.validation.FieldMatch;
+
+@Data
+@FieldMatch
+public class UserRegistrationRequestDto {
+    @Email
+    private String email;
```
Comment (vovasalyha) [inline src/main/java/mate/academy/bookstore/dto/UserRegistrationRequestDto.java]: What if I pass null here?

File: src/main/java/mate/academy/bookstore/dto/UserRegistrationRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    @NotNull
+    private String password;
+
+    @NotNull
+    private String repeatPassword;
+
+    @NotNull
```
Comment (vovasalyha) [inline src/main/java/mate/academy/bookstore/dto/UserRegistrationRequestDto.java]: What if I pass empty string here? Also, what if I pass string which is longer then allowed length in DB?

File: src/main/java/mate/academy/bookstore/exception/CustomGlobalExceptionHandler.java [hunk 4]
```diff
@@ -53,4 +53,13 @@ public ResponseEntity<Object> handleEntityNotFoundException(Exception ex) {
         body.put("errors", ex.getMessage());
         return new ResponseEntity<>(body, HttpStatus.NOT_FOUND);
     }
+
+    @ExceptionHandler(RegistrationException.class)
+    public ResponseEntity<Object> handleRegistrationException(Exception ex) {
+        Map<String, Object> body = new LinkedHashMap<>();
```
Comment (vovasalyha) [inline src/main/java/mate/academy/bookstore/exception/CustomGlobalExceptionHandler.java:59]: A lot of code duplication. Extract this responsed building process into separate method or create some class which can be DTO for error response.

**Review by vovasalyha — APPROVED** [review]
Looks good, but see a couple of comments.

**Review by vovasalyha — CHANGES_REQUESTED** [review]
Looks good, but see a couple of comments.

File: pom.xml (hunk 1 above)
Comment (bhdnchui) [reply pom.xml]: The question is still without an answer...

**Review by bhdnchui — APPROVED** [review]
Good job, but please answer @vovasalyha question)
